src/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php

```diff
@@ -278,39 +284,56 @@ public function create_item( $request ) {
 		require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
 		require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
 
-		$slug = $request['slug'];
+		$language_packs = array();
 
-		// Verify filesystem is accessible first.
-		$filesystem_available = $this->is_filesystem_available();
-		if ( is_wp_error( $filesystem_available ) ) {
-			return $filesystem_available;
-		}
+		if ( isset( $request['slug'] ) ) {
+			$slug = $request['slug'];
 
-		$api = plugins_api(
-			'plugin_information',
-			array(
-				'slug'   => $slug,
-				'fields' => array(
-					'sections'       => false,
-					'language_packs' => true,
-				),
-			)
-		);
+			// Verify filesystem is accessible first.
+			$filesystem_available = $this->is_filesystem_available();
+			if ( is_wp_error( $filesystem_available ) ) {
+				return $filesystem_available;
+			}
```

Comment:
This should be before the isset() checks. ie. It should execute even if the `url` parameter is used.